aiohttp/connector.py

```diff
@@ -834,9 +842,15 @@ def _create_direct_connection(self, req,
                     transp.close()
                     if not self._cleanup_closed_disabled:
                         self._cleanup_closed_transports.append(transp)
-                    raise ServerFingerprintMismatch(
+                    last_exc = ServerFingerprintMismatch(
                         expected, got, host, port)
+                    continue
+
             return transp, proto
+        else:
+            assert last_exc is not None
```

Comment:
Just drop the line.
Check doesn't point on potential problem, next line will fail enyway (raise None is forbidden in Python 3).